tests/selenium/pixel_tracking_test.py

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env python
+# -*- coding: UTF-8 -*-
+
+import unittest
+import pbtest
+
+class pixelTrackingTesting(pbtest.PBSeleniumTest):
+	"""Tests for the cookie pixel tracking heuristic included in heuristicblocking.js
+		- loads gitcdn resource which places a tracking cookie on page then creates an img tag
+		- img tag makes a src request carrying a substring of that tracking cookie
+		- tracking domain is caught by pixel tracking heuristic, snitch map entry is updated
+	"""
+
+	def test_that_tracker_is_caught(self):
+		TESTING_URL = ("https://gitcdn.link/repo/ablanathtanalba/pixelTrackingTestingResource/master/rsrc.html")
+
+		CHECK_SNITCH_MAP_FOR_ENTRY = (
+				"return chrome.extension.getBackgroundPage()."
+				"badger.storage.snitch_map.getItem('github.com').includes('gitcdn.link');"
+		)
+
+		self.load_url(TESTING_URL)
+		self.load_url(self.options_url)
+
+		# github will already appear in a pretrained badger instance for appearing on fontawesome.com
```

Comment:
Better to try to guarantee we detected cookie sharing tracking just now (and it wasn't already detected from seed data, nor that we didn't detect some other kind of tracking instead).

You could clear seed data before running the test to eliminate the pre-trained possibility.

To ensure we indeed detected pixel cookie sharing, you could first load a version of the test page that doesn't apply pixel cookie sharing (maybe via a query string flag), verify that tracking was not detected, and then reload the page, this time with pixel cookie sharing enabled, and verify that now tracking was detected.

Reply:
Cleared up this and [this](https://github.com/EFForg/privacybadger/pull/2425#discussion_r316770343) in my most recent local commit, including the proposed URLs for the testing resources when I get write access to the server.